### ISSUE TITLE
Absolute mean scores giving incorrect importances

### ISSUE THREAD

**Issue by malakar-soham** [interpretml/interpret#311]
Hi devs,
I'm having some confusions reading the graphs so I'll give a context and the confusion I'm facing

### Context
I'm working on a NLP , multiclass problem and feeding tokens using TF-IDF as the feature vector to a OneVsRest classifier wrapped over an EBM. The OneVsRest is creating as many EBM models as the classes but the feature vector for all of their training is the same.

### Observation:
This is the overall importance summary and let us consider the features 490 and 600:
![image](https://user-images.githubusercontent.com/38937943/146543992-429527fa-bc11-4ebd-b4b3-6f6db152b2da.png)

The influence of feature 490 looks like this:
![image](https://user-images.githubusercontent.com/38937943/146544154-55952236-337c-4575-a8da-987561b99900.png)

The influence of feature 600 looks like this:
![image](https://user-images.githubusercontent.com/38937943/146544187-0520ab63-1df4-48d3-8a95-02f331ce6c58.png)

The observation is according to the overall summary, feature 490 looks to be more important than 600 but on opening its influence on the model, it looks like 600 predicts the class better, the x-axis being the TF-IDF weighted document term matrix. Sometimes, the overall importance summary has top features which are present in the feature vector but with a value 0 for the class we are training.

### Questions
- What are `scores (y-axis)` in the feature summary and what is the best range for it to be in?
- How to read the overall importance summary considering the aforementioned anomaly ?

**Comment by interpret-ml** [comment]
Hi @malakar-soham, 

Good question! The overall summary is the _mean absolute_ value of the scores across the training set. Another way to think of it is "on average across the training dataset, how much does this feature contribute to a single prediction in absolute terms?". These scores are represented in the same units as the y-axis of the feature graphs, so for a classification problem, it would be in logits.

In the case of your example, the reason feature 490 has higher importance than feature 600 is because there are significantly more non-zero valued samples in feature 490:

![image](https://user-images.githubusercontent.com/50031429/146570932-982cf5d6-4618-45c8-8772-f5771ddcfb25.png)

With both features, a value of 0 will have a very small effect on the model prediction, but a non-zero value will have a large absolute effect (both large positive and large negative effects are considered important by our measure). Therefore, because there are significantly more non-zero samples in feature 490, it gets a correspondingly higher rank. 

I want to caveat all this by saying that our choice of importance measure is just one of many viable options. If you want to specifically capture extreme effects, even if they only affect relatively few samples, you may want to consider something like max() - min() on each graph. In the future we'd like to offer alternative methods of feature importance calculation directly in the package.

Hope this helps!
-InterpretML Team

**Comment by malakar-soham** [comment]
## Part 1
Thanks for the detailed explanation but the concept of score for continuous data is still a bit confusing for me. From your explanation, I can understand that the score is an absolute measure so even a high negative score with more non-negative examples will be considered more important than a high positive score with less non-negative examples.  
But for this statement:
>These scores are represented in the same units as the y-axis of the feature graphs.

- [ ] Could you please explain or provide a paper on how the values of the feature graphs are calculated, I want to get a rough  idea on how the scores are being calculated, since it is different for separate EBMs though the feature vector is same.


## Part 2
There's also some confusion regarding the local score explainability.
The following is a sample local explainable example:
![image](https://user-images.githubusercontent.com/38937943/146960554-42163ff2-2187-47d5-ab0c-625dcb314a66.png)

### Part 2 : Inference
```
I can understand that:
1 - Binary classification score (0/1)
3 - Feature name
4 - TF-IDF weighted document term score
```
- [ ] Could you please provide a bit more info on what the point 2 and point 5 represent

## Part 3
From the following 3 images, it can be observed that the first 2 images have the same features in the y-axis whereas the last image has slightly different features affecting it. 
![Image1](https://user-images.githubusercontent.com/38937943/146961737-52f1331e-4186-47d3-bbfb-d135e5f7c6dc.png)
Image1
![Image2](https://user-images.githubusercontent.com/38937943/146961403-0f1b4c9e-a256-407c-9637-15129e91f1ea.png)
Image2
![Image3](https://user-images.githubusercontent.com/38937943/146961361-6c58ac11-edb4-4ed1-bca1-01d969a46861.png)
Image3

### Part 3 : Inference
Correct me if I'm wrong, my understanding is that for image 3, the prediction is 0 because the features marked in blue is negative and the features marked in orange is positive for that example. As score being 4*5 (where 4 and 5 is discussed in Part 2) for each feature, is negative when summed up for all features, the prediction comes out to be negative.
Similarly, for image 1 the overall score is positive so the prediction is 1.
Also, we can see that the features 2 to 13 are always positive for image 1 and 2 but they turn negative when there are values for those features as seen in image 3.

Now, considering the feature vector is same for all 3 images:
- [ ] Is my inference of Part 3 correct?
- [ ] Why in image 2, feature 14 is present but there is no score for it whereas the non-present features have scores?
- [ ] Why are the positively affecting features changing it's polarity when those features are present in the example?
- [ ] On what metrics are the features being ranked ?

**Comment by Tejamr** [comment]
Hi malakar-soham, based on my understanding there is no specific method or metric for calculating Mean absolute score of each feature in EBM Global importance . Correct me if I am wrong .
Moreover,if any metrics or mathematical formulae there for calculating those scores ? Please let me know  if there . I stuck there for a long time .
